packages/components/tree-v2/src/composables/useTree.ts

```diff
@@ -88,31 +88,33 @@ export function useTree(
     const expandedKeys = expandedKeySet.value
     const hiddenKeys = hiddenNodeKeySet.value
     const flattenNodes: TreeNode[] = []
-    const nodes = (tree.value && tree.value.treeNodes) || []
-    function traverse() {
-      const stack: TreeNode[] = []
-      for (let i = nodes.length - 1; i >= 0; --i) {
-        stack.push(nodes[i])
+    const nodes = tree.value?.treeNodes || []
+
+    // Stack to simulate recursion, each item contains [nodes, index]
+    const stack: [TreeNode[], number][] = [[nodes, 0]]
```

Comment:
The data structure `[TreeNode[], number][]` appears to be over-design. I think `TreeNode[]` is just right for the function implementation.

Reply:
The [TreeNode[], number][] structure tracks our position in the tree, aiding navigation between related items. It handles large trees efficiently, particularly for partial tree operations. It's not redundant, but a deliberate choice to improve performance for complex tree manipulations.

Reply:
Without position tracking, I haven't hit any performance issues. What cases could there be a noticeable performance improvement compared to the following implementation? 

```ts
   const nodes = tree.value?.treeNodes || []

    const stack: TreeNode[] = []
    for (let i = nodes.length - 1; i >= 0; --i) {
      stack.push(nodes[i])
    }
    while (stack.length) {
      const node = stack.pop()!
      if (hiddenKeys.has(node.key)) continue

      flattenNodes.push(node)
      if (node.children && expandedKeys.has(node.key)) {
        for (let i = node.children.length - 1; i >= 0; --i) {
          stack.push(node.children[i])
        }
      }
    }
```

Reply:
After re-running performance tests with your code, I found that the difference is less than 100ms when expanding 20,000 tree nodes. Thank you for the code review